test/server/http/admin_test.cc

```diff
@@ -293,5 +294,57 @@ TEST(PrometheusStatsFormatter, FormattedTags) {
   EXPECT_EQ(expected, actual);
 }
 
+TEST(PrometheusStatsFormatter, statsAsPrometheus) {
+
+  // create two clusters and two gauges with metric name being same same and
+  // different tag names. In the end there only two metric names
+  // should be inserted, one for gauge and one for counter
+
+  Stats::TagExtractorImpl tag_extractor("cluster_name", "^cluster\\.((.+?)\\.)");
+  std::string name = "cluster.test_cluster_1.upstream_cx_total";
+  std::vector<Stats::Tag> tags;
+  Stats::Tag tag1 = {"a.tag-name", "a.tag-value"};
+  tags.push_back(tag1);
+  Stats::HeapRawStatDataAllocator alloc_;
+  Stats::RawStatData* data = alloc_.alloc(name);
+  Stats::CounterSharedPtr c1(
+      new Stats::CounterImpl(*data, alloc_, std::string(name), std::move(tags)));
+  std::list<Stats::CounterSharedPtr> counters;
+  counters.push_back(c1);
+
+  std::string name1 = "cluster.test_cluster_1.upstream_cx_total";
+  std::vector<Stats::Tag> tags1;
+  Stats::Tag tag2 = {"another_tag_name", "another_tag-value"};
+  tags1.push_back(tag2);
+  Stats::HeapRawStatDataAllocator alloc1_;
```

Comment:
@pitiwari I think the issue is a little subtle, but I think this is because the allocators are being destructed before the stats themselves.  

Fix: try declaring a single allocator as the first line in this test (to ensure it is the last thing to be destructed), and then use that allocator to allocate the data for all of the stats you create below.

Quick explanation of what's happening: the two lists `counters` and `gauges` are created before some of the allocators used to allocate the data for the `CounterImpl`s and `GuageImpl`s that they own.  This will delay the destruction of these stats until after their allocators are destructed, which means that the `CounterImpl` and `GaugeImpl` destructors will try to call methods on an object that has already been destructed.  The rule with these allocators is that they must outlive everything they allocate.